src/nodes.coffee

```diff
@@ -2351,47 +2351,25 @@ exports.Assign = class Assign extends Base
     if olen is 1 and obj instanceof Expansion
       obj.error 'Destructuring assignment has no target'
 
+    # Count all `Splats`: [a, b, c..., d, e]
+    splats = (i for obj, i in objects when obj instanceof Splat)
+    # Count all `Expansions`: [a, b, ..., c, d]
+    expans = (i for obj, i in objects when obj instanceof Expansion)
+    # Combine splats and expansions and sort them so we can show error if needed.
+    splatsAndExpans = [splats..., expans...].sort()
```

Comment:
Why sort?

Reply:
In case there are more than one `Splat` and/or `Expansion` I use `sort` so I can show the first error.
So, presume this example: `[a, b..., ..., c..., d, e]`
FIrst I collect indexes of all splats (`splats = [1, 3]`), and then expansions (`expans = [2]`).
First splat (`b...`, index 1) is valid, and I want to show the next error (`...`, index 2).
`[splats..., expans...].sort()` will put them in order.

Reply:
This could be:

```coffee
if splats.length isnt 0 and expans.length isnt 0
  # Combine and sort here, then throw error
```

No need to sort if we’re not throwing an error.

Reply:
Nice. I moved `sort()` after the condition.